Recipe edits for styled-components and Emotion (#455)
storybookjs/frontpage: src/content/recipes/styled-components.md

This post will explain how to:

1. 🔌 Setup `GlobalStyle`
2. 🧱 Use styled-components in your components
3. 💅 Use a theme in your stories
4. 🎨 Switch betweens themes in a click

If you’d like to see the example code of this recipe, check out the [example repository](https://github.com/Integrayshaun/styled-components-recipe) on GitHub. Let's get started!

![Completed styled-components example with theme switcher](https://user-images.githubusercontent.com/18172605/208312563-875ca3b0-e7bc-4401-a445-4553b48068ed.gif)

## How to setup `GlobalStyle`

UIs often have a set of global styles that are applied to every component like CSS resets, `font-size`, `font-family`, and colors.

In styled-components, use the [`createGlobalStyle`](https://styled-components.com/docs/api#createglobalstyle) API to scope styles globally instead of locally (which is the library's default behavior).

> Open `.storybook/previews.js` and create a `GlobalStyle` that includes a `font-family`. Then apply it to all stories via a [decorator](/docs/react/writing-stories/decorators).

Reviewer: ```suggestion
Open `.storybook/preview.js` and create a `GlobalStyle` component which includes a `font-family`. Then apply it to all stories via a [decorator](/docs/react/writing-stories/decorators).
```

```js
// .storybook/preview.js

const GlobalStyle = createGlobalStyle`
  body {
    font-family: "Nunito Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
  }
`;

const withGlobalStyle = (Story) => (
  <>
    <GlobalStyle />
    <Story />
  </>
);

export const decorators = [withGlobalStyle];
```

<div class="aside">

> If you already have `GlobalStyle` in your app, you can import it into `.storybook/previews.js` instead of creating it anew.

Reviewer: ```suggestion
If you already have `GlobalStyle` in your app, you can import it into `.storybook/preview.js` instead of creating it anew.
```

  },
};

> export const decorators = [withGlobalStyle, withTheme];

Reviewer: If we flip the order here, you can then access `theme` inside `GlobalStyle`. Perhaps we should demonstrate that, too? Or maybe just a comment explaining why the order matters? Something like this:

```suggestion
// Decorators run in the order they are provided. By placing `withTheme` before `withGlobalStyle`, the GlobalStyle component can access the `theme` via context
export const decorators = [withTheme, withGlobalStyle];
```

Author: I think describing this behavior may be too nuanced for a basic recipe, especially given that our `withGlobalStyle` doesn't use `theme`.

Reviewer: That's fair. I was concerned about that balance, too. Let's at least not set them up for failure and flip the order?

```suggestion
export const decorators = [withTheme, withGlobalStyle];
```

Reviewer 2: I thought that they were applied left to right? withGlobalStyles then withTheme would give globaltyles access to the theme

```

> Now, components made with styled-components will get the theme through the `theme` prop. Let's update the example components to use the theme.

Reviewer: Oh, in conjunction with \<previous comment>, we could add an aside after this paragraph, saying that this includes the GlobalStyle component?

Author: Addressed this [here](https://github.com/storybookjs/frontpage/pull/455/commits/c1fbd29d84c350bac9e519f9c6f09bc635e56c91). What do you think?

Reviewer: That's a good addition, but not quite what I was referring to here.

Author: Chatted to KG in Slack, going to ignore this
